src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -139,8 +146,10 @@ - (void)finishWithError:(NSError *)errorOrNil {
   _self = nil;
 
   // If there were still request messages coming, stop them.
-  _requestWriter.state = GRXWriterStateFinished;
-  _requestWriter = nil;
+  @synchronized(_requestWriter) {
```

Comment:
> I was going by this: http://rykap.com/objective-c/2015/05/09/synchronized.html. 

Please don't depend on undocumented behavior. 'undocumented behavior' is defined as 'behavior that has not been documented by Apple'

It looks like you  do want 
@synchronized(_requestWriter) {
  _requestWriter.state = GRXWriterStateFinished;
}
_requestWriter = nil;

as long as all objects that look at the same requestWriter @synchronize on that requestWriter - races are things like:
if (_requestWriter.state != GRXWriterStateXXX) {
   _requestWriter.state = GRXWriterStateYYY;
}

An if () {} like this would have to be inside a @synchronized(_requestWriter) { }